src/schedule/ucc_schedule_pipelined.h

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+#ifndef UCC_SCHEDULE_PIPELINED_H_
+#define UCC_SCHEDULE_PIPELINED_H_
+#include "components/base/ucc_base_iface.h"
+
+#define UCC_SCHEDULE_FRAG_MAX_TASKS 8
+
+typedef struct ucc_schedule_pipelined ucc_schedule_pipelined_t;
+
+#define UCC_SCHEDULE_PIPELINED_MAX_FRAGS 4
```

Comment:
question: at UCC level.  it should it make sense to use term "FRAG",  is this works only for fragmnetation (or) can we be used for any generic task pipeline?

Reply:
I'm not sure what do you mean? What is pipelining if you don't fragment the collective?